core/src/main/java/org/elasticsearch/script/ScriptModule.java

```diff
@@ -56,24 +54,55 @@ public static ScriptModule create(Settings settings, Environment environment,
         scriptEngineServices.add(nativeScriptEngineService);
         List<ScriptContext.Plugin> plugins = scriptPlugins.stream().map(x -> x.getCustomScriptContexts()).filter(Objects::nonNull)
                 .collect(Collectors.toList());
-        return new ScriptModule(settings, environment, resourceWatcherService, scriptEngineServices, plugins);
+        List<TemplateService.Backend> templateBackends = scriptPlugins.stream().map(x -> x.getTemplateBackend())
+                .filter(Objects::nonNull).collect(Collectors.toList());
+        TemplateService.Backend templateBackend;
+        switch (templateBackends.size()) {
+        case 0:
+            templateBackend = null;
+            break;
+        case 1:
+            templateBackend = templateBackends.get(0);
+            break;
+        default:
+            throw new IllegalArgumentException("Elasticsearch only supports a single template backend but was started with ["
+                    + templateBackends + "]");
+        }
+        return new ScriptModule(settings, environment, resourceWatcherService, scriptEngineServices, plugins, templateBackend);
     }
 
+    private final ScriptSettings scriptSettings;
+    private final ScriptService scriptService;
+    private final TemplateService templateService;
+
     /**
      * Build {@linkplain ScriptEngineService} and {@linkplain ScriptContext.Plugin}.
      */
     public ScriptModule(Settings settings, Environment environment,
                         ResourceWatcherService resourceWatcherService, List<ScriptEngineService> scriptEngineServices,
-                        List<ScriptContext.Plugin> customScriptContexts) {
+                        List<ScriptContext.Plugin> customScriptContexts, @Nullable TemplateService.Backend templateBackend) {
         ScriptContextRegistry scriptContextRegistry = new ScriptContextRegistry(customScriptContexts);
         ScriptEngineRegistry scriptEngineRegistry = new ScriptEngineRegistry(scriptEngineServices);
-        scriptSettings = new ScriptSettings(scriptEngineRegistry, scriptContextRegistry);
+        ScriptMetrics scriptMetrics = new ScriptMetrics();
+        // Note that if templateBackend is null this won't register any settings for it
+        scriptSettings = new ScriptSettings(scriptEngineRegistry, templateBackend, scriptContextRegistry);
```

Comment:
It is useful for this to be able to handle a null `templateBackend` so we let this stay null here.